factory_reset/reset.proto

```diff
@@ -0,0 +1,41 @@
+// This file defines a gNOI API used for factory reseting a Target.
+syntax = "proto3";
+
+package gnoi.factory_reset;
+
+import "github.com/openconfig/gnoi/types/types.proto";
+
+option (types.gnoi_version) = "0.1.0";
+
+// The FactoryReset service exported by Targets.
+service FactoryReset {
+  // The Initiate RPC allows the Client to instruct the Target to immediately
+  // clean all existing state and boot the Target in the same condition as it is
+  // shipped from factory. State includes storage, configuration, logs,
+  // certificates and licenses.
+  //
+  // It is optional for the Target to support rolling back the OS to the same
+  // version shipped from factory. If not supported, the running OS should be
+  // used.
+  //
+  // It is optional for the Target to zero fill permanent storage where state
+  // data is stored.
+  rpc Initiate(InitiateRequest) returns (InitiateResponse);
+}
+
+message InitiateRequest {
```

Comment:
are you interested in having a scheduled version where reset does not happen immediately?

Reply:
It is a good idea, however I cannot come up with a use case for it. =)